src/main/java/sirius/biz/tenants/TenantUserManager.java

```diff
@@ -506,45 +511,58 @@ public UserInfo findUserByCredentials(@Nullable WebContext ctx, String user, Str
-            auditLog.negative("AuditLog.externalLoginRequired")
-                    .causedByUser(account.getUniqueName(), account.getUserAccountData().getLogin().getUsername())
-                    .forUser(account.getUniqueName(), account.getUserAccountData().getLogin().getUsername())
-                    .forTenant(String.valueOf(account.getTenant().getId()),
-                               account.getTenant().getValue().getTenantData().getName())
-                    .log();
+            completeAuditLogForUser(auditLog.negative("AuditLog.externalLoginRequired"), account);
             throw Exceptions.createHandled().withNLSKey("UserAccount.externalLoginMustBePerformed").handle();
         }
 
         LoginData loginData = account.getUserAccountData().getLogin();
         if (acceptApiTokens && Strings.areEqual(password, loginData.getApiToken())) {
-            auditLog.neutral("AuditLog.apiTokenLogin")
-                    .causedByUser(account.getUniqueName(), account.getUserAccountData().getLogin().getUsername())
-                    .forUser(account.getUniqueName(), account.getUserAccountData().getLogin().getUsername())
-                    .forTenant(String.valueOf(account.getTenant().getId()),
-                               account.getTenant().getValue().getTenantData().getName())
-                    .log();
+            completeAuditLogForUser(auditLog.neutral("AuditLog.apiTokenLogin"), account);
             return result;
         }
 
-        if (loginData.checkPassword(password, defaultSalt)) {
-            auditLog.neutral("AuditLog.passwordLogin")
-                    .causedByUser(account.getUniqueName(), account.getUserAccountData().getLogin().getUsername())
-                    .forUser(account.getUniqueName(), account.getUserAccountData().getLogin().getUsername())
-                    .forTenant(String.valueOf(account.getTenant().getId()),
-                               account.getTenant().getValue().getTenantData().getName())
-                    .log();
+        LoginData.PasswordVerificationResult pwResult = loginData.checkPassword(password);
+        if (pwResult != LoginData.PasswordVerificationResult.INVALID) {
+            completeAuditLogForUser(auditLog.neutral("AuditLog.passwordLogin"), account);
+
+            if (pwResult == LoginData.PasswordVerificationResult.VALID_NEEDS_RE_HASH) {
+                return updatePasswordHashing(result, password);
+            }
+
             return result;
         }
 
-        auditLog.negative("AuditLog.loginRejected")
-                .forUser(account.getUniqueName(), account.getUserAccountData().getLogin().getUsername())
-                .forTenant(String.valueOf(account.getTenant().getId()),
-                           account.getTenant().getValue().getTenantData().getName())
-                .log();
+        completeAuditLogForUser(auditLog.negative("AuditLog.loginRejected"), account);
```

Comment:
This would result in a rejection in log caused by the user. This might me confusing to some people. ("I did not try to login but why does it state here that I did so?")